nexus/src/db/datastore/volume.rs

```diff
@@ -166,15 +166,164 @@ impl DataStore {
         Ok(())
     }
 
+    /// Get a copy of the Volume from the database.
+    /// This action (getting a copy) will increase the generation number
+    /// of Volumes of the VolumeConstructionRequest::Volume type that have
+    /// sub_volumes of the VolumeConstructionRequest::Region type.
+    /// This generation number increase is required for Crucible to support
+    /// crash consistency.
```

Comment:
Is there somewhere that I can read about what's happening on the Crucible side? I don't understand how the generation number, as it currently exists, is supporting crash consistency.

Specifically, what is the entity / operation that "fails" if it doesn't have a high-enough generation number? In this PR, I can only see us unconditionally bumping an integer.

Reply:
The [storage RFD has more details](https://rfd.shared.oxide.computer/rfd/0177#_reconciliation) but the decision about generation number is made when propolis activates a Volume it has constructed based on the `VolumeConstructionRequest` it has received from Nexus.  None of the generation number work is in Propolis or Crucible yet, as we have to be sure that Nexus is giving Propolis correct generation numbers first.